tasks/dev.py

```diff
@@ -193,13 +193,104 @@ def migrate(context: Context, database: str = INFRAHUB_DATABASE):
+def update_docker_compose(context: Context, docker_file: Optional[str] = "docker-compose.yml"):
+    """Update docker-compose.yml with the current version from pyproject.toml."""
+    import re
+
+    import toml
+
+    version = toml.load("pyproject.toml")["tool"]["poetry"]["version"]
+    version_pattern = r"registry.opsmill.io/opsmill/infrahub:\$\{VERSION:-[\d\.\-a-zA-Z]+\}"
+
+    def replace_version(match):
+        return f"registry.opsmill.io/opsmill/infrahub:${{VERSION:-{version}}}"
+
+    with open(docker_file, "r", encoding="utf-8") as file:
+        docker_compose = file.read()
+
+    updated_docker_compose = re.sub(version_pattern, replace_version, docker_compose)
+
+    with open(docker_file, "w", encoding="utf-8") as file:
+        file.write(updated_docker_compose)
+
+    print(f"{docker_file} updated with version {version}")
+
+
+def get_enum_mappings():
+    """Extracts enum mappings dynamically."""
+    from infrahub.config import StorageDriver, TraceExporterType, TraceTransportProtocol
+    from infrahub.database.constants import DatabaseType
+
+    enum_mappings = {}
+
+    for enum_class in [DatabaseType, StorageDriver, TraceExporterType, TraceTransportProtocol]:
+        for item in enum_class:
+            enum_mappings[item] = item.value
+
+    return enum_mappings
+
+
+def update_docker_compose_env_vars(
+    env_vars: list[str],
+    env_defaults: Dict[str, Any],
+    enum_mappings: Dict[Any, str],
+    docker_file: Optional[str] = "docker-compose.yml"
+) -> None:
+    """Update the docker-compose.yml file with the environment variables."""
+    from infrahub.config import StorageDriver, TraceExporterType, TraceTransportProtocol
+    from infrahub.database.constants import DatabaseType
+
+    with open(docker_file, "r", encoding="utf-8") as file:
+        docker_compose = file.readlines()
+
+    in_infrahub_config_section = False
+
+    for i, line in enumerate(docker_compose):
+        if line.strip().startswith("x-infrahub-config: &infrahub_config"):
+            in_infrahub_config_section = True
+            continue
+        if line.strip().startswith("services:"):
+            in_infrahub_config_section = False
+
+        if in_infrahub_config_section and line.strip().startswith("INFRAHUB_"):
+            var_name, existing_value = line.split(":", 1)
+            var_name = var_name.strip()
+            existing_value = existing_value.strip().strip('"')
+            if var_name in env_vars:
+                default_value = env_defaults.get(var_name, None)
+                if default_value is not None:
+                    # Convert bool to lowercase strings
+                    if isinstance(default_value, bool):
+                        default_value = str(default_value).lower()
+                    # Check if the default value is an enum-like value and replace it with its actual value
+                    elif isinstance(default_value, (DatabaseType, StorageDriver, TraceExporterType, TraceTransportProtocol)):
```

Comment:
I think you can directly do
```suggestion
                    elif isinstance(default_value, Enum):
```

and use `default_value.value`